src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3795,13 +3876,119 @@ export class StudyViewPageStore
         );
     }
 
+    private sampleIdToGenePanelId = observable.map<string, string>();
+
+    readonly genePanelData = remoteData<{
```

Comment:
All this logic should be done backend. Need to create new study-view API to support through backend. Also need to support it in StudyViewFilter object